dhcpv4/ztpv4/ztp.go

```diff
@@ -15,39 +17,59 @@ type VendorData struct {
 
 var errVendorOptionMalformed = errors.New("malformed vendor option")
 
-// ParseVendorData will try to parse dhcp4 options looking for more
-// specific vendor data (like model, serial number, etc).
-func ParseVendorData(packet *dhcpv4.DHCPv4) (*VendorData, error) {
+func parseVIVC(vd *VendorData, packet *dhcpv4.DHCPv4) error {
```

Comment:
I am not a fan of modifying the parameter, can it return a `*VendorData` instead and make it consistent with the `ParseVendorData` function definition?

Reply:
So, return vd explicitly instead of modifying it there?

Reply:
Yes, and not take a `*VendorData` as parameter. Is it needed as parameter?

Reply:
If we want to modify it there, then just pass a pointer (the way its implemented now). I honestly like that approach since its cleaner. 

Otherwise, we need to initialize a `vd := &VendorData{}` in each of the functions and modify the local version of vd and return it. I can go either way :D